// pages/Article/ArticleContainer.js
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { fetchArticle, fetchComments, createComment } from './actions';
import Article from './Article';
import { articleSelector, commentsSelector } from './selector';

function ArticleContainer({ match, fetchArticle, article, fetchComments, comments, createComment, currentUser }) {

    const { slug } = match.params;

    useEffect(() => {
        fetchArticle(slug);
        fetchComments(slug);
    }, [fetchArticle, fetchComments, slug]);

    return (
        <Article
            {...article}
            comments={comments}
            createComment={createComment}
            slug={slug}
            currentUser={currentUser}
        />
    );
}

ArticleContainer.propTypes = {
    fetchArticle: PropTypes.func.isRequired,
    article: PropTypes.object.isRequired,
    fetchComments: PropTypes.func.isRequired,
    comments: PropTypes.object.isRequired,
    createComment: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
    article: articleSelector(state),
    comments: commentsSelector(state),
    currentUser: state.auth.currentUser,
});

const mapDispatchToProps = {
    fetchArticle,
    fetchComments,
    createComment,
};

export default connect(mapStateToProps, mapDispatchToProps)(ArticleContainer);
