// pages/settings/index.js
import React from 'react';
import { connect } from 'react-redux';
import { Row, Col, Form, Input, Button } from 'antd';

import Layout from 'components/Layout';

const { TextArea } = Input;

const validateMessages = {
    required: 'is required!',
    types: {
        email: 'is not validate email!',
        number: 'is not a validate number!',
    },
    number: {
        range: '${label} must be between ${min} and ${max}',
    },
};

function Settings() {
    return (
        <Layout>
            <section className="settings">
                <Row>
                    <Col span={12} offset={6}>
                        <Form validateMessages={validateMessages}>
                            <Form.Item>
                                <Input
                                    placeholder="URL изображения профиля"
                                />
                            </Form.Item>
                            <Form.Item
                                name="username"
                                rules={[
                                    {
                                        required: true,
                                        message: 'Please input your Username!',
                                    },
                                ]}
                            >
                                <Input
                                    placeholder="Имя пользователя"
                                />
                            </Form.Item>
                            <Form.Item>
                                <TextArea
                                    autoSize={{ minRows: 8 }}
                                    placeholder="Краткая биография о вас"
                                />
                            </Form.Item>
                            <Form.Item
                                name="email"
                                rules={[
                                    {
                                        type: 'email',
                                        required: true,
                                        message: 'Email is not validate email!',
                                    },
                                ]}
                            >
                                <Input
                                    type="email"
                                    placeholder="Электронный адрес"
                                />
                            </Form.Item>
                            <Form.Item
                                name="password"
                                rules={[
                                    {
                                        required: true,
                                        message: 'Please input your Password!',
                                    },
                                ]}
                            >
                                <Input
                                    type="password"
                                    placeholder="Новый пароль"
                                />
                            </Form.Item>
                            <Form.Item>
                                <Button
                                    type="primary"
                                    htmlType="submit"
                                >
                                    Обновить настройки
                                </Button>
                            </Form.Item>
                        </Form>
                    </Col>
                </Row>
            </section>
        </Layout>
    );
}

const mapStateToProps = (state) => ({

});

const mapDispatchToProps = {

};

export default connect(mapStateToProps, mapDispatchToProps)(Settings);
