// reducers/article.js
const initialState = {
    isFetching: false,
    isFetchingComments: false,
    data: {
        author: {
            username: '',
        }
    },
    comments: [],
    error: null,
    status: null,
};

export default (state = initialState, { type, payload }) => {

    switch (type) {
        case 'FETCH_ARTICLE_REQUEST':
            return { ...state, isFetching: true };
        case 'FETCH_ARTICLE_SUCCESS':
            return { ...state, isFetching: false, data: payload.article };
        case 'FETCH_ARTICLE_FAILURE':
            return { ...state, isFetching: false, error: payload.error, status: payload.status };

        case 'FETCH_ARTICLE_COMMENTS_REQUEST':
            return { ...state, isFetchingComments: true };
        case 'FETCH_ARTICLE_COMMENTS_SUCCESS':
            return { ...state, isFetchingComments: false, comments: payload.comments };
        case 'FETCH_ARTICLE_COMMENTS_FAILURE':
            return { ...state, isFetchingComments: false, error: payload.error, status: payload.status };

        case 'FETCH_LIKE_REQUEST':
        case 'FETCH_UNLIKE_REQUEST':
            return { ...state };
        case 'FETCH_LIKE_SUCCESS':
        case 'FETCH_UNLIKE_SUCCESS':
            return {
                ...state,
                data: {
                    ...state.data,
                    favorited: payload.article.favorited,
                    favoritesCount: payload.article.favoritesCount,
                }
            };
        case 'FETCH_LIKE_FAILURE':
        case 'FETCH_UNLIKE_FAILURE':
            return { ...state, error: payload };
        default:
            return state;
    }
};
