// reducers/auth.js
import {
    LOGIN,
    REGISTER,
    LOGIN_PAGE_UNLOADED,
    REGISTER_PAGE_UNLOADED,
    ASYNC_START,
    UPDATE_FIELD_AUTH
} from 'constants/types';

const initialState = {
    isFetching: false,
    user: {
        username: null,
    },
    errors: null,
    isAuthenticated: false,
    username: '',
    email: 'seryozha.baleyko@gmail.com',
    password: 'gfhjkm1997',
};

export default (state = initialState, action) => {

    switch (action.type) {
        case 'POST_LOGIN_REQUEST':
        case 'POST_REGISTER_REQUEST':
        case 'FETCH_AUTHENTICATION_REQUEST':
            return { ...state, isFetching: true };
        case 'POST_LOGIN_SUCCESS':
        case 'POST_REGISTER_SUCCESS':
        case 'FETCH_AUTHENTICATION_SUCCESS':
            return { ...state, isFetching: false, user: action.payload, isAuthenticated: true };
        case 'POST_LOGIN_FAILURE':
        case 'POST_REGISTER_FAILURE':
        case 'FETCH_AUTHENTICATION_FAILURE':
            return { ...state, isFetching: false, errors: action.payload };

        case LOGIN:
        case REGISTER:
            return {
                ...state,
                isFetching: false,
                errors: action.error ? action.payload.errors : null,
            };
        case LOGIN_PAGE_UNLOADED:
        case REGISTER_PAGE_UNLOADED:
            return {};
        case ASYNC_START:
            if (action.subtype === LOGIN || action.subtype === REGISTER) {
                return { ...state, isFetching: true };
            }
            break;
        case UPDATE_FIELD_AUTH:
            return { ...state, [action.key]: action.value };
        default:
            return state;
    }
};
