// pages/Home/components/Articles.js
import React from 'react';
import PropTypes from 'prop-types';
import { Alert, Pane, Spinner } from 'evergreen-ui';

import Article from './Article';

import './Articles.scss';

function Articles({ isFetching, items, itemsCount, errorMessage }) {

    return (
        <section className="articles">
            {errorMessage && (
                <Alert
                    intent="danger"
                    title={errorMessage}
                    style={{
                        margin: '30px 0',
                    }}
                />
            )}
            {isFetching && (
                <Pane display="flex" alignItems="center" justifyContent="center" height={300}>
                    <Spinner/>
                </Pane>
            )}
            <div className="articles-list">
                {items.length > 0 && items.map((article) => (
                    <Article key={article.slug} {...article} />
                ))}
            </div>
            {!!itemsCount && (
                'pagination'
            )}
        </section>
    );
}

Articles.propTypes = {
    isFetching: PropTypes.bool.isRequired,
    items: PropTypes.array.isRequired,
    itemsCount: PropTypes.number.isRequired,
    errorMessage: PropTypes.string,
};

export default Articles;
