// pages/Home/actionTypes.js
export const GET_ARTICLES_REQUEST = 'GET_ARTICLES_REQUEST';
export const GET_ARTICLES_SUCCESS = 'GET_ARTICLES_SUCCESS';
export const GET_ARTICLES_FAILURE = 'GET_ARTICLES_FAILURE';

export const GET_ARTICLES_BY_AUTHOR_REQUEST = 'GET_ARTICLES_BY_AUTHOR_REQUEST';
export const GET_ARTICLES_BY_AUTHOR_SUCCESS = 'GET_ARTICLES_BY_AUTHOR_SUCCESS';
export const GET_ARTICLES_BY_AUTHOR_FAILURE = 'GET_ARTICLES_BY_AUTHOR_FAILURE';

export const GET_ARTICLES_FAVORITES_REQUEST = 'GET_ARTICLES_FAVORITES_REQUEST';
export const GET_ARTICLES_FAVORITES_SUCCESS = 'GET_ARTICLES_FAVORITES_SUCCESS';
export const GET_ARTICLES_FAVORITES_FAILURE = 'GET_ARTICLES_FAVORITES_FAILURE';

export const GET_ARTICLES_FEED_REQUEST = 'GET_ARTICLES_FEED_REQUEST';
export const GET_ARTICLES_FEED_SUCCESS = 'GET_ARTICLES_FEED_SUCCESS';
export const GET_ARTICLES_FEED_FAILURE = 'GET_ARTICLES_FEED_FAILURE';

export const GET_TAGS_REQUEST = 'GET_TAGS_REQUEST';
export const GET_TAGS_SUCCESS = 'GET_TAGS_SUCCESS';
export const GET_TAGS_FAILURE = 'GET_TAGS_FAILURE';

export const APPLY_TAG_FILTER_REQUEST = 'APPLY_TAG_FILTER_REQUEST';
export const APPLY_TAG_FILTER_SUCCESS = 'APPLY_TAG_FILTER_SUCCESS';
export const APPLY_TAG_FILTER_FAILURE = 'APPLY_TAG_FILTER_FAILURE';

export const POST_LIKE_REQUEST = 'POST_LIKE_REQUEST';
export const POST_LIKE_SUCCESS = 'POST_LIKE_SUCCESS';
export const POST_LIKE_FAILURE = 'POST_LIKE_FAILURE';

export const DELETE_LIKE_REQUEST = 'DELETE_LIKE_REQUEST';
export const DELETE_LIKE_SUCCESS = 'DELETE_LIKE_SUCCESS';
export const DELETE_LIKE_FAILURE = 'DELETE_LIKE_FAILURE';
