// pages/Home/Home.js
import React from 'react';
import { NavLink, useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Pane, Spinner, Alert } from 'evergreen-ui';
// import Pagination from 'react-router-pagination';

import Cover from 'components/Cover';
import Article from 'components/Article';
import Tags from './components/Tags';

function Home({ articles, tags, fetchTag }) {

    const { pageNumber = 1 } = useParams();

    return (
        <div className="main">
            <Cover/>
            <div className="container">
                <div className="home">
                    <div style={{ height: '65px' }}></div>
                    <div style={{
                        display: 'grid',
                        gridTemplateColumns: '1fr 272px',
                        columnGap: '25px',
                    }}>
                        <div>
                            <ul className="tabs">
                                <li className="tabs-item">
                                    <NavLink to="/" className="tabs-link" exact>
                                        Global Feed
                                    </NavLink>
                                </li>
                                <li className="tabs-item">
                                    <NavLink to="/feed" className="tabs-link">
                                        Your Feed
                                    </NavLink>
                                </li>
                            </ul>
                            <div>
                                {articles.isFetching && (
                                    <Pane display="flex" alignItems="center" justifyContent="center" height={400}>
                                        <Spinner size={32}/>
                                    </Pane>
                                )}
                                {articles.errorMessage && (
                                    <Alert
                                        appearance="card"
                                        intent="danger"
                                        title={articles.errorMessage}
                                        marginTop={25}
                                        width="100%"
                                    />
                                )}
                                {articles.data.map((article) => (
                                    <Article key={article.slug} {...article} />
                                ))}
                                {/*<Pagination
                                    totalPages={500}
                                    pageNumber={pageNumber}
                                />*/}
                            </div>
                        </div>
                        <div>
                            <Tags {...tags} fetchTag={fetchTag} />
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}

Home.propTypes = {
    articles: PropTypes.object.isRequired,
    tags: PropTypes.object.isRequired,
};

export default Home;
