// pages/Home/HomeContainer.js
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Home from './Home';
import { fetchTags } from 'pages/Home/actions/tags';
import { articlesSelector, tagsSelector } from './selectors';
import { fetchArticles } from 'pages/Home/actions/articles';
import { fetchTag } from 'pages/Home/actions/tag';

function HomeContainer({ fetchArticles, articles, fetchTags, tags, fetchTag }) {

    useEffect(() => {
        fetchArticles();
        fetchTags();
    }, [fetchArticles, fetchTags]);

    return (
        <Home
            articles={articles}
            tags={tags}
            fetchTag={fetchTag}
        />
    );
}

HomeContainer.propTypes = {
    fetchArticles: PropTypes.func.isRequired,
    fetchTags: PropTypes.func.isRequired,
    articles: PropTypes.shape({
        isFetching: PropTypes.bool.isRequired,
        data: PropTypes.array.isRequired,
    }).isRequired,
    tags: PropTypes.shape({
        isFetching: PropTypes.bool.isRequired,
        data: PropTypes.array.isRequired,
    }).isRequired,
};

const mapStateToProps = (state) => ({
    articles: articlesSelector(state),
    tags: tagsSelector(state),
});

const mapDispatchToProps = {
    fetchTag,
    fetchTags,
    fetchArticles,
};

export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer);
