// pages/Home/index.js
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link, NavLink, Switch, Route } from 'react-router-dom';
import PropTypes from 'prop-types';

import { fetchTags } from 'actions/tags';
import { fetchArticles } from 'actions/articles';
import Layout from 'components/Layout';
import Article from 'components/Article';
import YourFeed from 'components/YourFeed';

import './home.scss';

function Tags({ tags }) {

    const handleClick = (event) => {
        event.preventDefault();

    };

    return (
        <div>
            {tags.map((tag) => (
                <Link
                    to="#"
                    key={tag}
                    onClick={handleClick}
                >
                    {tag}
                </Link>
            ))}
        </div>
    );
}

function Home({ fetchArticles, fetchTags, isFetching, articles, tags, error }) {

    useEffect(() => {
        fetchArticles();
    }, [fetchArticles]);

    useEffect(() => {
        fetchTags();
    }, [fetchTags]);

    return (
        <Layout>
            <section className="banner" style={{ margin: '50px 0' }}>banner</section>
            <section className="home">
                <div className="content">
                    <ul className="tabs">
                        <li className="tabs-item">
                            <NavLink className="tabs-link" to="/" exact>Global Feed</NavLink>
                        </li>
                        <li className="tabs-item">
                            <NavLink className="tabs-link" to="/feed">Your Feed</NavLink>
                        </li>
                    </ul>
                    <div className="articles-list">
                        <Switch>
                            <Route path="/" exact>
                                {articles.map((article) => (
                                    <Article key={article.slug} {...article} />
                                ))}
                            </Route>
                            <Route path="/feed" component={YourFeed} />
                        </Switch>
                    </div>
                </div>
                <div className="sidebar">
                    <p>Popular Tags</p>
                    <Tags tags={tags} />
                </div>
            </section>
        </Layout >
    );
}

Home.propTypes = {
    fetchArticles: PropTypes.func.isRequired,
    fetchTags: PropTypes.func.isRequired,
    isFetching: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
    isFetching: { tags: state.tags.isFetching, },
    tags: state.tags.tags,
    articles: state.articles.articles,
    error: { tags: state.tags.error, }
});

const mapDispatchToProps = {
    fetchTags,
    fetchArticles,
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
