// pages/Home/index.js
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import Layout from 'components/Layout';
import { fetchTags } from 'actions/tags';
import { fetchArticles, fetchArticlesFeed } from 'actions/articles';

import './home.scss';

function Tags({ tags }) {

    const handleClick = (event) => {
        event.preventDefault();

    };

    return (
        <div>
            {tags.map((tag) => (
                <Link
                    to="#"
                    key={tag}
                    onClick={handleClick}
                >
                    {tag}
                </Link>
            ))}
        </div>
    );
}

function Home({ fetchArticles, fetchArticlesFeed, fetchTags, isFetching, tags, error }) {

    useEffect(() => {
        fetchArticles();
    }, [fetchArticles]);

    useEffect(() => {
        fetchArticlesFeed();
    }, [fetchArticlesFeed]);

    useEffect(() => {
        fetchTags();
    }, [fetchTags]);

    return (
        <Layout>
            <section className="home">
                <h1>Home</h1>
                <div className="sidebar">
                    <p>Popular Tags</p>
                    <Tags tags={tags} />
                </div>
            </section>
        </Layout>
    );
}

Home.propTypes = {
    fetchArticles: PropTypes.func.isRequired,
    fetchArticlesFeed: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
    isFetching: { tags: state.tags.isFetching, },
    tags: state.tags.tags,
    error: { tags: state.tags.error, }
});

const mapDispatchToProps = {
    fetchTags,
    fetchArticles,
    fetchArticlesFeed,
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
