// pages/Article/components/ArticleSingle/index.js
import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'evergreen-ui';
import marked from 'marked';

import Author from 'components/Author';
import Like from 'components/Like';

import './articleSingle.scss';

function ArticleSingle({ slug, title, body = '', author, createdAt, favorited, favoritesCount, tagList = [], currentUser }) {

    const markup = { __html: marked(body) };

    // const isModify = currentUser && currentUser.username === author.username;

    const handleClickDelete = (event) => {
        event.preventDefault();

    };

    return (
        <article className="article-single">
            <header className="article-single-heading">
                <h1 className="article-single-title">{title}</h1>
                <div style={{
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'space-between',
                }}>
                    <Author
                        {...author}
                        createdAt={createdAt}
                    />
                    <div style={{
                        display: 'flex',
                        alignItems: 'center',
                    }}>
                        <Button is="a" href={`/article/${slug}/edit`} marginRight={16} iconBefore="edit">
                            Edit Article
                        </Button>
                        <Button
                            marginRight={16}
                            iconBefore="delete"
                            intent="danger"
                            onClick={handleClickDelete}
                        >
                            Delete Article
                        </Button>
                        <Like
                            slug={slug}
                            favorited={favorited}
                            favoritesCount={favoritesCount}
                        />
                    </div>
                </div>
            </header>
            <div className="article-single-body" dangerouslySetInnerHTML={markup}></div>
            <ul className="article-single-tagList">
                {tagList.length > 0 && tagList.map((tag) => (
                    <Button key={tag}>{tag}</Button>
                ))}
            </ul>
        </article>
    );
}

ArticleSingle.propTypes = {
    body: PropTypes.string,
    tagList: PropTypes.any,
};

export default ArticleSingle;
