// actions/article.js
import { API } from 'constants/index';

const token = localStorage.getItem('jwt');

export const fetchArticle = (slug) => async (dispatch) => {

    dispatch({
        type: 'FETCH_ARTICLE_REQUEST',
    });

    try {
        const url = `${API}/articles/${slug}`;
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_ARTICLE_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_ARTICLE_FAILURE',
            payload: error,
        });
    }
};

export const fetchArticleComments = (slug) => async (dispatch) => {

    dispatch({
        type: 'FETCH_ARTICLE_COMMENTS_REQUEST',
    });

    try {
        const url = `${API}/articles/${slug}/comments`;
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_ARTICLE_COMMENTS_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_ARTICLE_COMMENTS_FAILURE',
            payload: error,
        });
    }
};
