// actions/articles.js
import { API } from 'constants/index';

const token = localStorage.getItem('jwt');

export const fetchArticles = (limit = 10, offset) => async (dispatch) => {

    dispatch({
        type: 'FETCH_ARTICLES_REQUSET',
    });

    try {
        const url = `${API}/articles?limit=${limit}&offset=${offset ? offset * limit : 0}`;
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_ARTICLES_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_ARTICLES_FAILURE',
            payload: error,
        });
    }
};

export const fetchArticlesByAuthor = (author = '', page = 0, limit = 5) => async (dispatch) => {

    dispatch({
        type: 'FETCH_ARTICLES_BY_AUTHOR_REQUEST',
    });

    try {
        const url = `${API}/articles?author=${encodeURIComponent(author)}&limit=${limit}&offset=${page ? page * limit : 0}`;
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_ARTICLES_BY_AUTHOR_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_ARTICLES_BY_AUTHOR_FAILURE',
            payload: error,
        });
    }
};

export const fetchYourFeedArticles = (limit = 10, offset = 0) => async (dispatch) => {

    dispatch({
        type: 'FETCH_ARTICLES_FEED_REQUEST',
    });

    try {
        const url = `${API}/articles/feed?limit=${limit}&offset=${offset}`;
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_ARTICLES_FEED_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_ARTICLES_FEED_FAILURE',
            payload: error,
        });
    }
};

export const like = (slug) => async (dispatch) => {

    dispatch({
        type: 'FETCH_LIKE_REQUEST',
    });

    try {
        const url = `${API}/articles/${slug}/favorite`;
        const options = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_LIKE_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_LIKE_FAILURE',
            payload: error,
        });
    }
};

export const unlike = (slug) => async (dispatch) => {

    dispatch({
        type: 'FETCH_UNLIKE_REQUEST',
    });

    try {
        const url = `${API}/articles/${slug}/favorite`;
        const options = {
            method: 'DELETE',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_UNLIKE_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_UNLIKE_FAILURE',
            payload: error,
        });
    }
};
