// actions/auth.js
import { API } from 'constants/index';

export const signUp = (username, email, password) => async (dispatch) => {

    dispatch({
        type: 'POST_REGISTER_REQUEST',
    });

    const url = `${API}/users`;
    const options = {
        method: 'POST',
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json;charset=utf-8',
        },
        body: JSON.stringify({ user: { username, email, password } }),
    };

    fetch(url, options)
        .then((response) => response.json())
        .then((result) => {
            if (result.errors) {
                dispatch({
                    type: 'POST_REGISTER_FAILURE',
                    payload: result.errors,
                });
            } else {
                localStorage.setItem('jwt', result.user.token);

                dispatch({
                    type: 'POST_REGISTER_SUCCESS',
                    payload: result.user,
                });
            }
        });
};

export const signIn = (email, password) => async (dispatch) => {

    dispatch({
        type: 'POST_LOGIN_REQUEST',
    });

    const url = `${API}/users/login`;
    const options = {
        method: 'POST',
        headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json;charset=utf-8',
        },
        body: JSON.stringify({ user: { email, password } }),
    };

    fetch(url, options)
        .then((response) => response.json())
        .then((result) => {
            if (result.errors) {
                dispatch({
                    type: 'POST_LOGIN_FAILURE',
                    payload: result.errors,
                });
            } else {
                localStorage.setItem('jwt', result.user.token);

                dispatch({
                    type: 'POST_LOGIN_SUCCESS',
                    payload: result.user,
                });
            }
        });
};

export const getProfileFetch = () => async (dispatch) => {

    const token = localStorage.getItem('jwt');

    if (token) {

        dispatch({
            type: 'FETCH_AUTHENTICATION_REQUEST'
        });

        try {
            const url = `${API}/user`;
            const options = {
                method: 'GET',
                headers: {
                    Accept: 'application/json',
                    'Content-Type': 'application/json;charset=utf-8',
                    'Authorization': `Token ${token}`
                },
            };
            const response = await fetch(url, options);
            const result = await response.json();

            dispatch({
                type: 'FETCH_AUTHENTICATION_SUCCESS',
                payload: result.user,
            });
        } catch (error) {
            localStorage.removeItem('jwt');

            dispatch({
                type: 'FETCH_AUTHENTICATION_FAILURE',
                payload: error,
            });
        }

    }
};

export const logoutUser = () => ({
    type: 'LOGOUT_USER'
});
