// actions/like.js
import { API } from 'constants/index';

const token = localStorage.getItem('jwt');

export const like = (slug) => async (dispatch) => {

    dispatch({
        type: 'FETCH_LIKE_REQUEST',
    });

    try {
        const url = `${API}/articles/${slug}/favorite`;
        const options = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_LIKE_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_LIKE_FAILURE',
            payload: error,
        });
    }
};

export const unlike = (slug) => async (dispatch) => {

    dispatch({
        type: 'FETCH_UNLIKE_REQUEST',
    });

    try {
        const url = `${API}/articles/${slug}/favorite`;
        const options = {
            method: 'DELETE',
            headers: {
                'Content-Type': 'application/json;charset=utf-8',
                'authorization': token ? `Token ${token}` : '',
            },
        };
        const response = await fetch(url, options);
        const result = await response.json();

        dispatch({
            type: 'FETCH_UNLIKE_SUCCESS',
            payload: result,
        });
    } catch (error) {
        dispatch({
            type: 'FETCH_UNLIKE_FAILURE',
            payload: error,
        });
    }
};
