// components/Author/index.js
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Avatar } from 'evergreen-ui';

import './author.scss';

function Author({ username, bio, image, following, createdAt }) {

    return (
        <div className="author">
            <Link className="author-thumbnail" to={`/@${username}`}>
                <Avatar
                    src={image}
                    size={32}
                />
            </Link>
            <div className="author-body">
                <h5 className="author-title">
                    <Link to={`/@${username}`}>
                        {username}
                    </Link>
                </h5>
                <span className="author-date">{new Date(createdAt).toDateString()}</span>
            </div>
        </div>
    );
}

Author.propTypes = {
    username: PropTypes.string,
    bio: PropTypes.string,
    image: PropTypes.string,
    following: PropTypes.bool,
    createdAt: PropTypes.string,
};

export default Author;
