// components/Author/index.js
import React from 'react';
import PropTypes from 'prop-types';

import './author.scss';
import { Link } from 'react-router-dom';

function Author({ username, bio, image, following, createdAt }) {
    return (
        <div className="author">
            <div className="author-thumbnail">
                <Link to={`/@${username}`}>
                    <img src={image} alt={username} />
                </Link>
            </div>
            <div className="author-body">
                <h5 className="author-title">
                    <Link to={`/@${username}`}>
                        {username}
                    </Link>
                </h5>
                <span className="author-date">{new Date(createdAt).toDateString()}</span>
            </div>
        </div>
    );
}

Author.propTypes = {
    username: PropTypes.string.isRequired,
    bio: PropTypes.string,
    image: PropTypes.string.isRequired,
    following: PropTypes.bool,
    createdAt: PropTypes.string.isRequired,
};

export default Author;
