// pages/Profile/index.js
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

import Layout from 'components/Layout';
import { fetchProfile, follow, unfollow } from 'actions/profile';

import './profile.scss';

const account = {
    width: '60%',
    marginTop: '60px',
    marginLeft: 'auto',
    marginRight: 'auto',
};

function Profile({ match, fetchProfile, profile, currentUser, follow, unfollow }) {

    const username = match.params.username;
    const isUser = currentUser && profile.username === currentUser.username;

    const handleFollowAndUnfollowClick = (event) => {
        event.preventDefault();

        if (profile.following) {
            unfollow(profile.username);
        } else {
            follow(profile.username);
        }
    };

    useEffect(() => {
        fetchProfile(username);
    }, [fetchProfile, username]);

    const classes = classNames('button', {
        'primary': !profile.following,
        'default': profile.following,
    });

    return (
        <Layout>
            <section className="profile">
                <div className="user" style={account}>
                    <div className="user-thumbnail">
                        <img src={profile.image} alt={profile.username} />
                    </div>
                    <div className="user-body">
                        <h1>{profile.username}</h1>
                        <p>{profile.bio}</p>
                        {isUser ? (
                            <Link to={`/@${username}/edit`}>
                                Редактировать профиль
                            </Link>
                        ) : (
                                <button
                                    className={classes}
                                    type="button"
                                    onClick={handleFollowAndUnfollowClick}
                                >
                                    {profile.following ? 'Unfollow' : 'Follow'}
                                </button>
                            )}
                    </div>
                </div>
                <div className="articles">
                    <div className="articles-heading">
                        <ul>
                            <li>
                                <Link to="#">Мои статьи</Link>
                            </li>
                            <li>
                                <Link to="">Любимые статьи</Link>
                            </li>
                        </ul>
                    </div>
                    <div className="articles-list">
                        article
                    </div>
                </div>
            </section>
        </Layout>
    );
}

const mapStateToProps = (state) => ({
    isFetching: state.profile.isFetching,
    profile: state.profile,
    currentUser: state.auth.user,
});

const mapDispatchToProps = {
    fetchProfile,
    follow,
    unfollow,
};

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
