// pages/Profile/index.js
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Link, NavLink, Switch, Route } from 'react-router-dom';
import classNames from 'classnames';

import { fetchProfile, follow, unfollow } from 'actions/profile';
import { fetchArticlesByAuthor } from 'actions/articles';
import Layout from 'components/Layout';
import Article from 'components/Article';

import './profile.scss';

const account = {
    width: '60%',
    marginTop: '60px',
    marginLeft: 'auto',
    marginRight: 'auto',
};

function Profile({ match, fetchProfile, fetchArticlesByAuthor, profile, currentUser, follow, unfollow, articles }) {

    const { url, params } = match;
    const { username } = params;
    const isUser = currentUser && profile.username === currentUser.username;

    const handleFollowAndUnfollowClick = (event) => {
        event.preventDefault();
        if (profile.following) {
            unfollow(profile.username);
        } else {
            follow(profile.username);
        }
    };

    useEffect(() => {
        fetchProfile(username);
        fetchArticlesByAuthor(username);
    }, [fetchProfile, fetchArticlesByAuthor, username]);

    const classes = classNames('button', {
        'primary': !profile.following,
        'default': profile.following,
    });

    return (
        <Layout>
            <section className="profile">
                <div className="user" style={account}>
                    <div className="user-thumbnail">
                        <img src={profile.image} alt={profile.username} />
                    </div>
                    <div className="user-body">
                        <h1>{profile.username}</h1>
                        <p>{profile.bio}</p>
                        {isUser ? (
                            <Link to={`/@${username}/edit`}>
                                Редактировать профиль
                            </Link>
                        ) : (
                                <button
                                    className={classes}
                                    type="button"
                                    onClick={handleFollowAndUnfollowClick}
                                >
                                    {profile.following ? 'Unfollow' : 'Follow'}
                                </button>
                            )}
                    </div>
                </div>
                <div className="articles">
                    <div className="articles-heading">

                    </div>
                    <ul className="tabs">
                        <li className="tabs-item">
                            <NavLink className="tabs-link" to={`${url}`} exact>Мои статьи</NavLink>
                        </li>
                        <li className="tabs-item">
                            <NavLink className="tabs-link" to={`${url}/favorites`}>Любимые статьи</NavLink>
                        </li>
                    </ul>
                    <div className="articles-list">
                        <Switch>
                            <Route path={`${url}`} exact>
                                {articles.map((article) => (
                                    <Article key={article.slug} {...article} />
                                ))}
                            </Route>
                            <Route path={`${url}/favorites`}>
                                favorites
                            </Route>
                        </Switch>
                    </div>
                </div>
            </section>
        </Layout>
    );
}

const mapStateToProps = (state) => ({
    isFetching: state.profile.isFetching,
    profile: state.profile,
    currentUser: state.auth.user,
    articles: state.articles.articles,
});

const mapDispatchToProps = {
    fetchProfile,
    follow,
    unfollow,
    fetchArticlesByAuthor,
};

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
